numpy/core/src/multiarray/item_selection.c

```diff
@@ -2289,8 +2289,15 @@ PyArray_Nonzero(PyArrayObject *self)
         }
         else {
             npy_intp j;
+            npy_intp added_count = 0;
             for (j = 0; j < count; ++j) {
                 if (nonzero(data, self)) {
+                    if (++added_count > nonzero_count) {
+                        PyErr_SetString(PyExc_RuntimeError,
```

Comment:
I do not believe you can set the errorstate without the GIL? You should probably just break and check if something odd happened after the fact. If we error at too many items read, I think we may as well error at too few items read.

Reply:
> I think we may as well error at too few items read.

I was going to suggest it, and was surprised to see it was already fixed, so thanks for noticing!